src/Model/ImageLoading.py

```diff
@@ -607,25 +649,80 @@ def calculate_matrix(img_ds):
-          position[1]],
-         [orientation[2] * dist_row, orientation[5] * dist_col, 0,
-          position[2]],
-         [0, 0, 0, 1]]
+        [
+            [orientation[0] * dist_row, orientation[3] * dist_col, 0, position[0]],
+            [orientation[1] * dist_row, orientation[4] * dist_col, 0, position[1]],
+            [orientation[2] * dist_row, orientation[5] * dist_col, 0, position[2]],
+            [0, 0, 0, 1],
+        ]
     )
-    x = []
-    y = []
-    for i in range(0, img_ds.Columns):
-        i_mat = matrix_m * np.matrix([[i], [0], [0], [1]])
-        x.append(float(i_mat[0]))
 
-    for j in range(0, img_ds.Rows):
-        j_mat = matrix_m * np.matrix([[0], [j], [0], [1]])
-        y.append(float(j_mat[1]))
+    # Optimization attempt: Pre-allocate arrays
+    # x = [None] * img_ds.Columns
+    # y = [None] * img_ds.Rows
+    # for i in range(img_ds.Columns):
+    #     i_mat = matrix_m * np.matrix([[i], [0], [0], [1]])
+    #     x[i] = float(i_mat[0])
+
+    # for j in range(img_ds.Rows):
+    #     j_mat = matrix_m * np.matrix([[0], [j], [0], [1]])
+    #     y[j] = float(j_mat[1])
+
+    # return np.array(x), np.array(y)
+
+    # Optimization attempt: avoid repeated matrix multiplications
+    # Calculate base values (origin)
+    origin = matrix_m * np.matrix([[0], [0], [0], [1]])
+
+    # Calculate increments (first column of matrix_m for x, second column for y)
+    x_increment = matrix_m[0, 0]  # First column, first row element
+    y_increment = matrix_m[1, 1]  # Second column, second row element
+
+    # # Create arrays efficiently
+    # x = np.array(
+    #     [float(origin[0]) + i * float(x_increment) for i in range(img_ds.Columns)]
+    # )
+    # y = np.array(
+    #     [float(origin[1]) + j * float(y_increment) for j in range(img_ds.Rows)]
+    # )
+
+    # return x, y
+
+    # Optimization attempt: avoid repeated matrix multiplications
+    # and repeated scalar multiplications
+    # Pre-allocate arrays
+    # x = np.empty(img_ds.Columns, dtype=np.float64)
+    # y = np.empty(img_ds.Rows, dtype=np.float64)
+
+    # # Set initial values
+    # x[0] = float(origin[0])
+    # y[0] = float(origin[1])
+
+    # # Fill arrays using cumulative addition
+    # for i in range(1, img_ds.Columns):
+    #     x[i] = x[i - 1] + x_increment
+
+    # for j in range(1, img_ds.Rows):
+    #     y[j] = y[j - 1] + y_increment
+
+    # return x, y
+
+    # Optimization attempt: Create arrays using vectorized operations
+    # x = np.arange(img_ds.Columns, dtype=np.float64) * x_increment + float(origin[0])
+    # y = np.arange(img_ds.Rows, dtype=np.float64) * y_increment + float(origin[1])
+
+    # return x, y
+
+    # Optimization attempt: use numpy to generate linearly spaced arrays in one go.
+    # Calculate end points
+    x_end = float(origin[0]) + (img_ds.Columns - 1) * x_increment
+    y_end = float(origin[1]) + (img_ds.Rows - 1) * y_increment
+
+    # Generate linearly spaced arrays in one operation
```

Comment:
from counting (one, one thousand...) this seemed to be the fastest *inclusive* of the time spent on CT histogram.